### PR TITLE
make README an actual markdown file; add compiler directive needed for g...

### PR DESCRIPTION
**Pull request by air**
...enerics to work

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -9,6 +9,13 @@
     <build>
         <plugins>
             <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
```
Comment (elingg) [inline java/pom.xml]: Could we go up to 1.6 or 1.7 instead of Java 1.5?

**Comment by elingg** [comment]
:+1: Left one comment about Java Version, other than that, feel free to merge. LGTM

**Comment by air** [comment]
I thought about it, 1.5 is the minimum to get it to compile, so that's a
sensible setting (nobody actually has to use Java 1.5). It should be the
most compatible way to express this.

**Comment by elingg** [comment]
Makes sense. Thanks!

File: java/pom.xml (hunk 1 above)
Comment (jsancio) [inline java/pom.xml]: All those versions are "EOL" by Oracle. If we are going to update the version might as well update to 8.

Comment (elingg) [inline java/pom.xml]: This change is fine since it lets you conpile with 1.5 and up. It just means any version >=5.

**Pull request merged**